docs/developing_iree/profiling.md

```diff
@@ -6,256 +6,18 @@ system behavior in more depth, there are various ways to
-```
-
-The above compiles IREE with Tracy APIs so that IREE will stream profiling data
-back to Tracy when running. To collect and analyze these data, you can either
-use GUI or CLI tools. Tracy profiler is the GUI tool. You can find the
-Tracy manual on its [releases page](https://github.com/wolfpld/tracy/releases)
-for more details on Tracy itself.
-
-#### Building on Linux
-
-To build the profiler on Linux, you may need to install some external
-libraries. Some Linux distributions will require you to add a `lib` prefix and a
-`-dev`, or `-devel` postfix to library names. For example, you might see the
-error:
-
-```
-Package glfw3 was not found in the pkg-config search path.
-```
-
-and then you could try to install `libglfw3-dev`.
-
-Instructions to build Tracy profiler:
-
-```shell
-$ cd third_party/tracy/profiler/build/unix
-$ make release
-```
-
-### Using Tracy
-
-Launch the profiler UI and click connect to start waiting for a traced program
-to running. Now you can launch the IREE binary you want to trace and Tracy
-should connect automatically and stream data. For example:
-
-Compile a .mlir file using `iree-translate`:
-
-```shell
-$ build/iree/tools/iree-translate \
-  -iree-mlir-to-vm-bytecode-module \
-  -iree-hal-target-backends=vmla \
-  $PWD/iree/tools/test/simple.mlir \
-  -o /tmp/simple.vmfb
-```
-
-Run a compiled module once:
-
-```shell
-$ build/iree/tools/iree-run-module \
-  --module_file=/tmp/simple.vmfb \
-  --driver=vmla \
-  --entry_function=abs \
-  --function_inputs="i32=-2"
-```
-
-Benchmark a compiled module, running it many times:
-
-```shell
-$ build/iree/tools/iree-benchmark-module \
-  --module_file=/tmp/simple.vmfb \
-  --driver=vmla \
-  --entry_function=abs \
-  --function_inputs="i32=-2"
-```
-
-> Note:<br>
-> &nbsp;&nbsp;&nbsp;&nbsp;IREE binaries may finish running before even
-> connecting to Tracy. For such cases, you can set `TRACY_NO_EXIT=1` in the
-> environment to keep the IREE binary alive until Tracy connects to it.
-
-### Configuring Tracy
-
-Set IREE's `IREE_TRACING_MODE` value (defined in
-[iree/base/tracing.h](https://github.com/google/iree/blob/main/iree/base/tracing.h))
-to adjust which tracing features, such as allocation tracking and callstacks,
-are enabled.
-
-In order for Tracy to record detailed statistics via sampling, the program
-collecting data must be run using elevated permissions (Administrator on Windows,
-root on Linux, rooted Android device). See Tracy's user manual for more
-information.
+The IREE team primarily relies on the Tracy profiler. Consider using it by default unless a more specialized need arises. Refer to [profiling_with_tracy.md](./profiling_with_tracy.md).
 
 ## Vulkan GPU Profiling
 
-Tracy offers great insights into CPU/GPU interactions and Vulkan API usage
+[Tracy](./profiling_with_tracy.md) offers great insights into CPU/GPU
+interactions and Vulkan API usage
 details. However, information at a finer granularity, especially inside a
```

Comment:
nit: re-flow lines (wrapping at 80 characters, not cutting off early like here)

if you use Visual Studio Code, you can set
```json
    "editor.rulers": [
        80,
        100
    ],
```
in your settings (other editors/IDEs have similar settings) to help, or go all the way to an autoformatter. We discussed markdown formatting tools but didn't settle on an open source one that we liked enough :P

Reply:
Thanks for the tip! Applied!